pyproject.toml

```
@@ -101,6 +101,7 @@ norecursedirs = [
 ]
 filterwarnings = [
     "error::ResourceWarning",
+    "error:.*finalize.*:pytest.PytestUnraisableExceptionWarning",
```

```suggestion
    # Files left open by tests can trigger ResourceWarnings during test
    # teardown which otherwise would be converted back into a warning
    # by pytest. Turn these PytestUnraisableExceptionWarnings back
    # into errors
    "error::pytest.PytestUnraisableExceptionWarning",
```
Is the message wildcard needed?

---

I committed the suggestion, but why is this wildcard not a good thing?  I thought it should catch only specifically the exception that happens when cleanup fails. I don't really know the full scope of when an UnraisableExceptionWarning can happen

---

Honestly it's more for the prior PR that removed pytest-openfiles than this PR (where the finalize filter would work). Leaving a file open can result in a PytestUnraisableExceptionWarning that doesn't contain 'finalize'.

One reason related to this PR is that I'm not sure if pytest considers it's warning messages as part of it's stable API. I kind of doubt it so any pytest release might change the warning message. I expect that the exception class is part of the stable API.

---

good point RE the stability of the API and thanks for explaining your thinking